src/layout/core/borders/borders.rs

```diff
@@ -345,6 +369,13 @@ impl Borders {
     pub fn set_title(&mut self, title: String) {
         self.title = title;
     }
+
+    pub fn set_children(&mut self, titles: Vec<String>, index: usize) {
+        self.children = Some(Children{
```

Comment:
Put an assert in here to verify that the `index` is a valid index for the vector.

Also add that in the doc comment, so it's obvious to the user of the function to do that.